flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -286,29 +285,10 @@ public static void main(String[] args) throws Exception {
 			LOG.info("Cannot determine the maximum number of open file descriptors");
 		}
 
-		final Configuration configuration = loadConfiguration(args);
-
-		FileSystem.initialize(configuration, PluginUtils.createPluginManagerFromRootFolder(configuration));
-
-		SecurityUtils.install(new SecurityConfiguration(configuration));
-
-		try {
-			SecurityUtils.getInstalledContext().runSecured(new Callable<Void>() {
-				@Override
-				public Void call() throws Exception {
-					runTaskManager(configuration, ResourceID.generate());
-					return null;
-				}
-			});
-		} catch (Throwable t) {
-			final Throwable strippedThrowable = ExceptionUtils.stripException(t, UndeclaredThrowableException.class);
-			LOG.error("TaskManager initialization failed.", strippedThrowable);
-			System.exit(STARTUP_FAILURE_RETURN_CODE);
-		}
+		runTaskManagerSecurely(args, ResourceID.generate());
 	}
 
-	@VisibleForTesting
-	static Configuration loadConfiguration(String[] args) throws FlinkParseException {
+	public static Configuration loadConfiguration(String[] args) throws FlinkParseException {
```

Comment:
This change should be put together with another commit "[hotfix][yarn] Use public methods of TaskManagerRunner to simply YarnTaskExecutorRunner codes", because it is actually required by that commit, not the scope of this commit.

Reply:
Make sense. I will squash the two commit to one.